### PR TITLE
Stephen Florian - Jasmine Basics

### PR DESCRIPTION
**Pull request by coloradude**

### CODE REVIEW

File: grader.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  letterGrader: function(score){
+    grade = null;
+    if (score >= 90){grade = 'A'}
+    else if (score < 90 && score >=90){grade = 'B'}
+    else if (score < 80 && score >= 70){grade = 'C'}
+    else if (score < 70 && score >=60){grade = 'D'}
+    else {grade ='F'} 
+    return grade;
+  },
```
Comment (craftninja) [inline grader.js]: Nice code alignment for quick understanding. How could this be refactored to not need the && statements?

File: grader.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+module.exports = {
+
+  letterGrader: function(score){
+    grade = null;
```
Comment (craftninja) [inline grader.js]: What kind of variable is this?

File: grader.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    else {grade ='F'} 
+    return grade;
+  },
+
+  averageScore: function(array){
+    return array.reduce(function(a,b){return a + b;}) / array.length;
+  },
+
+  medianScore: function(array){
+    meanId = array.length / 2;
```
Comment (craftninja) [inline grader.js]: What kind of variable is this?

File: spec/grader_spec.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    expect(code.modeScore([82, 92, 75, 91, 92, 89, 95, 100, 86])).toEqual(92);
+  });
+});
+
+
+// describe('', function() {
+//   it('', function() {
+//     expect(code.XXX()).toEqual();
+//   });
+// });
```
Comment (craftninja) [inline spec/grader_spec.js]: This cruft should be deleted before commit.

File: grader.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  },
+
+  modeScore: function(array){
+    var occurenceCounter = [];
+    for (var i=0;i<array.length;i++){
+      occurenceCounter.push(array.filter(function(score){return score === array[i]}).length)
+    }
+    return array[occurenceCounter.indexOf(Math.max.apply(Math, occurenceCounter))];
+  }
+}
```
Comment (craftninja) [inline grader.js]: This seems a little convoluted... is there a way to make your logic more obvious for the next person to understand?

**Comment by craftninja** [comment]
Whitespacing looks great, all tests pass. Once you get something working, consider refactoring to make sure the code is quick to read and understand. Nice job!

File: grader.js (hunk 1 above)
Comment (coloradude) [inline grader.js]: I guess I could just use >= 80 = B, >=70 = C etc and the function will just cascade down until it hits a true case.

File: grader.js (hunk 2 above)
Comment (coloradude) [inline grader.js]: Supposed to be empty. I wasn't sure you could run comparisons against an empty variable.

File: spec/grader_spec.js (hunk 4 above)
Comment (coloradude) [inline spec/grader_spec.js]: Noted.

**Pull request closed without merging**